index.js

```diff
@@ -873,6 +878,49 @@ class Batch {
   }
 }
 
+class EntryWatcher extends ReadyResource {
+  constructor (bee, key) {
+    super()
+
+    bee._watchers.add(this)
+    this.bee = bee
+
+    this.key = key
+    this.node = null
+
+    this.ready().catch(safetyCatch)
+  }
+
+  async _open () {
+    this.node = await this.bee.get(this.key)
+  }
+
+  _close () {
+    this.bee._watchers.delete(this)
+  }
+
+  async _onappend () {
+    if (!this.opened) await this.ready()
+
+    let newNode
+    try {
+      newNode = await this.bee.get(this.key)
+    } catch (e) {
+      if (e.code === 'SNAPSHOT_NOT_AVAILABLE') {
+        // There was a truncate event before the get resolved
+        // So this handler will run again anyway
```

Comment:
just set newNode to null here and continue

Reply:
As mentioned, disregarding this (context: we're just ignoring this invocation, no need to set null here--it will figure out the correct node value next time `update` runs)